nft/src/lib.rs

```diff
@@ -857,12 +873,10 @@ pub mod pallet {
 		/// Must be called by registered enclaves.
 		#[pallet::weight(T::WeightInfo::add_secret_shard())]
 		pub fn add_secret_shard(origin: OriginFor<T>, nft_id: NFTId) -> DispatchResultWithPostInfo {
-			ensure!(false, Error::<T>::ComingSoon);
 			let who = ensure_signed(origin)?;
-			ensure!(
-				/* TODO is_registered_enclave(who) */ true,
-				Error::<T>::NotARegisteredEnclave
-			);
+
+			ensure!(Self::is_registered_enclave(&who), Error::<T>::NotARegisteredEnclave);
```

Comment:
Here you are calling is_registered_enclave and after you call get_enclave. It makes 2 reads of the same value, you should get it once, check it and use it